src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorSourceDocument.cs

```diff
@@ -10,6 +10,7 @@ namespace Microsoft.AspNetCore.Razor.Evolution
     internal class DefaultRazorSourceDocument : RazorSourceDocument
     {
         private MemoryStream _stream;
+        private string _content;
 
         public DefaultRazorSourceDocument(MemoryStream stream, Encoding encoding, string filename)
```

Comment:
This constructor should change. In the original implementation the `stream` was the buffer holding the data. In your example it is the `content`. We don't need both. Do the reading in the factory method before calling this ctor.